packages/flutter_tools/lib/src/android/gradle_utils.dart

```diff
@@ -17,41 +20,46 @@ import '../cache.dart';
 import '../globals.dart' as globals;
 import '../project.dart';
 import '../reporting/reporting.dart';
-import 'android_studio.dart';
 
-/// The environment variables needed to run Gradle.
-Map<String, String> get gradleEnvironment {
-  final Map<String, String> environment = Map<String, String>.of(globals.platform.environment);
-  if (javaPath != null) {
-    // Use java bundled with Android Studio.
-    environment['JAVA_HOME'] = javaPath;
-  }
-  // Don't log analytics for downstream Flutter commands.
-  // e.g. `flutter build bundle`.
-  environment['FLUTTER_SUPPRESS_ANALYTICS'] = 'true';
-  return environment;
-}
+const String _defaultGradleVersion = '6.7';
 
-/// Gradle utils in the current [AppContext].
-GradleUtils get gradleUtils => context.get<GradleUtils>();
+final RegExp _androidPluginRegExp = RegExp(r'com\.android\.tools\.build:gradle:(\d+\.\d+\.\d+)');
 
-/// Provides utilities to run a Gradle task,
-/// such as finding the Gradle executable or constructing a Gradle project.
+/// Provides utilities to run a Gradle task, such as finding the Gradle executable
+/// or constructing a Gradle project.
 class GradleUtils {
+  GradleUtils({
+    @required Platform platform,
+    @required Logger logger,
+    @required FileSystem fileSystem,
+    @required Cache cache,
+    @required OperatingSystemUtils operatingSystemUtils,
+  }) : _platform = platform,
+       _logger = logger,
+       _cache = cache,
+       _fileSystem = fileSystem,
+       _operatingSystemUtils = operatingSystemUtils;
+
+  final Cache _cache;
+  final FileSystem _fileSystem;
+  final Platform _platform;
+  final Logger _logger;
+  final OperatingSystemUtils _operatingSystemUtils;
+
   /// Gets the Gradle executable path and prepares the Gradle project.
   /// This is the `gradlew` or `gradlew.bat` script in the `android/` directory.
   String getExecutable(FlutterProject project) {
     final Directory androidDir = project.android.hostAppGradleRoot;
-    gradleUtils.injectGradleWrapperIfNeeded(androidDir);
```

Comment:
This line is sus. Basically, what this did is re-grabbed the global GradleUtils - the tests used this to test `getExecutable` against a mocked out `injectgradleWrapperIfNeeded`. I removed this functionality